cpp-ch/local-engine/Parser/CHColumnToSparkRow.cpp

```diff
@@ -68,41 +68,52 @@ ALWAYS_INLINE bool isBitSet(const char * bitmap, int32_t index)
 }
 
 static void writeFixedLengthNonNullableValue(
-    char * buffer_address, int64_t field_offset, const ColumnWithTypeAndName & col, int64_t num_rows, const std::vector<int64_t> & offsets)
+    char * buffer_address,
+    int64_t field_offset,
+    const ColumnWithTypeAndName & col,
+    size_t num_rows,
+    const std::vector<int64_t> & offsets,
+    const MaskVector & masks = nullptr)
 {
     FixedLengthDataWriter writer(col.type);
-    for (size_t i = 0; i < static_cast<size_t>(num_rows); i++)
-        writer.unsafeWrite(col.column->getDataAt(i), buffer_address + offsets[i] + field_offset);
+    for (size_t i = 0; i < num_rows; i++)
+    {
+        size_t row_idx = masks == nullptr ? i : masks->at(i);
```

Comment:
This may slow down for loop performance because of extra if-elses for fixed-length data types, better to add another function with masks